spotbugs/src/main/java/edu/umd/cs/findbugs/SAXBugCollectionHandler.java

```diff
@@ -573,7 +578,38 @@ private void parseBugInstanceContents(String qName, Attributes attributes) throw
         } else if ("UserAnnotation".equals(qName)) {
             // legacy from the cloud plugin stuff
         } else {
-            throw new SAXException("Unknown bug annotation named " + qName);
+            // @Anemone, add custom bug annotation types,
+            // which can be deserialized with 'fromXML' method in its class.
+            Method fromXML = qnameCache.computeIfAbsent(qName, k -> {
+                for (Plugin plugin : Plugin.getAllPlugins()) {
+                    Class<?> annotationClazz;
+                    try {
+                        // The qName should equal to its classname, so we can reflect the class by 'qName'.
+                        annotationClazz = plugin.getClassLoader().loadClass(k);
+                        return annotationClazz.getMethod("fromXML", String.class, Attributes.class);
+                    } catch (NoSuchMethodException | ClassCastException e) {
+                        e.printStackTrace();
+                    } catch (ClassNotFoundException ignored) {
+                        LOG.warn("{} not found in Plugin({})", k, plugin.getPluginId());
+                        // The current plugin classloader doesn't have the annotation class called 'qName', ignore.
+                    }
+                }
+                return null;
+            });
+            if (fromXML == null) {
+                throw new SAXException("Unknown bug annotation named " + qName);
+            }
+            try {
+                bugAnnotation = (BugAnnotation) fromXML.invoke(null, qName, attributes);
+            } catch (IllegalAccessException e) {
+                throw new SAXException("Factory method for " + qName + " is not accessible.", e);
+            } catch (InvocationTargetException e) {
+                if (e.getTargetException() instanceof Exception) {
+                    throw new SAXException("Factory method for " + qName + " threw an exception.", (Exception) e.getTargetException());
+                } else {
+                    throw new SAXException("Factory method for " + qName + " threw an exception:\n" + e.getTargetException().getStackTrace());
```

Comment:
LGTM reported [new finding here](https://lgtm.com/projects/g/spotbugs/spotbugs/rev/a77ab08634687b7791e902636996ab6184462693) (as the last concatenation argument will be converted to something like `[Ljava.lang.StackTraceElement;@6d06d69c` I presume).

Reply:
Thanks, I'll fix it at #1242